v3/pkg/application/application_darwin_delegate.m

```diff
@@ -17,6 +17,14 @@ - (void)themeChanged:(NSNotification *)notification {
         processApplicationEvent(EventApplicationDidChangeTheme, NULL);
     }
 }
+- (BOOL)applicationShouldHandleReopen:(NSNotification *)notification 
+                    hasVisibleWindows:(BOOL)flag {
+    if( hasListeners(EventApplicationShouldHandleReopen) ) {
+        processApplicationEvent(EventApplicationShouldHandleReopen, NULL);
```

Comment:
Would it be useful to pass the flag with the event?

Reply:
Updated. The `processApplicationEvent` now accept NSDirectory and will save to the `ApplicationEventContext`